### ISSUE TITLE
Redesign the lesson content page

### ISSUE THREAD

**Issue by flacial** [garageScript/c0d3-app#2321]
### Things to redesign

<details>
<summary>
Lesson parts nav
</summary>
<img width="453" alt="image" src="https://user-images.githubusercontent.com/35906419/190858048-f5caf6a1-117d-4bf4-a8fc-85faa4eb2ecd.png">
</details>

<details>
<summary>
Elements spacing & letter spacing
</summary>
<img width="966" alt="image" src="https://user-images.githubusercontent.com/35906419/190858103-2beae61f-88e8-40b3-8e90-c681dcd9606d.png">
<img width="1095" alt="image" src="https://user-images.githubusercontent.com/35906419/190858116-a1710e40-4f00-4859-88c4-66e66a7de9e1.png">
</details>

<details>
<summary>
Fonts
</summary>
The fonts are inconsistent. The should be `Inter` font

<img width="826" alt="image" src="https://user-images.githubusercontent.com/35906419/190858173-4d4f89ac-78af-4e6a-a11f-2eca8fb45ef8.png">
</details>

<details>
<summary>
Challengebar font, size, colors, weight, and letters spacing
</summary>
<img width="774" alt="image" src="https://user-images.githubusercontent.com/35906419/190858216-ba33de7e-e9e5-4c3a-8df2-d23d3564ba4d.png">
</details>

<details>
<summary>
Table of content
</summary>
It's too large
<img width="400" alt="image" src="https://user-images.githubusercontent.com/35906419/190858259-f0d1e6da-0b13-4149-ad74-2de7a9d5f88f.png">
</details>

<details>
<summary>
Text length
</summary>
It's hard to read lengthy text

<img width="1319" alt="image" src="https://user-images.githubusercontent.com/35906419/190858312-1b64eb21-b94d-49c9-81d9-441dbc0ea766.png">
</details>

---

We can use the [design mock](https://zpl.io/7wzQ7d8) we created in the past for this page. I think it would do the job.

**Comment by akshayishaan** [comment]
Is anyone working in this issue? I want to work on this issue. Can @flacial explain a bit more about it?

**Comment by flacial** [comment]
> Is anyone working in this issue? I want to work on this issue. Can @flacial explain a bit more about it?

No one is working on this issue. The main thing to do is to change the [page's](https://www.c0d3.com/curriculum/js0/primitive_data_and_operators) design to this [mock design](https://zpl.io/7wzQ7d8). Doing so will fix some UI issues that affect the student experience while studying (also polish the page).

**Comment by akshayishaan** [comment]
@flacial It was my bad.....in understanding your issue. Unfortunately i am not a designer. I thought you have the design and You need me to code this up. Anyways.....I am new here, exploring open source issues. Nice talking to you.

**Comment by anthonykhoa** [comment]
> I thought you have the design and You need me to code this up. 

From what I understand, the designs are ready and someone needs to code them up. I think you misunderstood @flacial's words

**Comment by SlyBouhafs** [comment]
After a bit of tweaking, thanks to @flacial for the help, here is the new lessons page redesign to match our new design, there are two variants, left TOC and right TOC, we can vote on which one we should implement.
<details> 
  <summary>Right TOC page:</summary>

![Image](https://user-images.githubusercontent.com/1410462/205298092-ba1f9931-f51a-4c2d-af12-148d73918d3b.png)

</details>

<details> 
  <summary>Right TOC scroll:</summary>

![Image](https://user-images.githubusercontent.com/1410462/205298865-87677e6c-72ca-4254-8f8e-438b6cfe7f06.png)

</details>

<details> 
  <summary>Left TOC page:</summary>

![Image](https://user-images.githubusercontent.com/1410462/205299093-41cea97e-dc38-4b2c-be5b-e0d531d17221.png)

</details>

<details> 
  <summary>Left TOC scroll:</summary>

![Image](https://user-images.githubusercontent.com/1410462/205299045-015dd8b8-59ff-4d24-b607-b29f9cea1678.png)

</details>

If you guys have any feedback, i would love to hear it.

**Comment by flacial** [comment]
@SlyBouhafs I vote for Left TOC page. It looks perfect!

**Comment by flacial** [comment]
I like the idea of having a container around the content (especially headers):

<img width="787" alt="image" src="https://user-images.githubusercontent.com/35906419/205481256-b253f72a-c257-4354-ae07-b472f34c92a5.png">

With the current design, things look like they're floating around and nothing to contain them (not pretty):

<img width="654" alt="image" src="https://user-images.githubusercontent.com/35906419/205481294-85b96119-cfe5-415d-bf68-084d09f312b3.png">

**Comment by SlyBouhafs** [comment]
> With the current design, things look like they're floating around and nothing to contain them (not pretty):
I agree, that header is mostly for students/mentors, we need to adapt to that it works with the admin pages too, i will work on it too.
